kythe/go/platform/vfs/vfs.go

```diff
@@ -63,6 +63,9 @@ type Writer interface {
 	// Create creates a new file for writing, as os.Create.
 	Create(ctx context.Context, path string) (io.WriteCloser, error)
 
+	// CreateTemp creates a new temp file returning the path to it.
+	CreateTemp(ctx context.Context, dir string, pattern string) (string, error)
```

Comment:
There isn't a correctness TOCTOU issue; the file gets created, so we "own" it. Regardless, I made the changes. I think it makes vfs a little weird (the first *os.File occurrence instead of some interface). I could introduce a new interface that extends WriteCloser and gives access to the name (required to use the other vfs operations) but I think that may be overkill.

Reply:
Adding a new type was what I had in mind (see my other comment in reply to your recent commit). It might indeed be overkill: But I think at very least if we want to return a path the documentation needs to explicitly spell out the expectations, e.g.

"An empty writable file is created and its path is returned; the caller is responsible for cleaning it up when it is no longer in use (either by removing or renaming it)." 

Also if the caller wants to rename, they'll need to make sure the eventual target directory is on the same filesystem as `dir`, otherwise the rename may not work. The implementation of `TempFile` can't enforce that, but someone using the file will need to know it.

Reply:
Added NamedWriteCloser